File: app/backend/src/hospital_ai/schemas/timeline.py
from datetime import datetime
from typing import Literal, Optional, Any
from pydantic import BaseModel
from .base import ApiSchema

class TimelineEventBase(ApiSchema):
    event_id: str
    timestamp: datetime
    type: Literal["chat", "document", "audit"]
    title: str
    body: str
    patient_id: Optional[str] = None
    metadata: dict[str, Any] = {}

class GlobalTimelineResponse(ApiSchema):
    events: list[TimelineEventBase]
    total_count: int

File: app/frontend/src/lib/api/timeline.ts
import { apiFetch } from "../api-client";

export interface TimelineEvent {
  event_id: string;
  timestamp: string;
  type: "chat" | "document" | "audit";
  title: string;
  body: string;
  patient_id?: string;
  metadata: Record<string, any>;
}

export interface TimelineResponse {
  events: TimelineEvent[];
  total_count: number;
}

export async function getGlobalTimeline(limit = 50, offset = 0): Promise<TimelineResponse> {
  return apiFetch(`/timeline?limit=${limit}&offset=${offset}`);
}
